src/prefect/infrastructure/process.py

```diff
@@ -247,6 +248,42 @@ def _get_environment_variables(self, include_os_environ: bool = True):
     def _base_flow_run_command(self):
         return [get_sys_executable(), "-m", "prefect.engine"]
 
+    def get_corresponding_worker_type(self):
+        return "process"
+
+    async def generate_work_pool_base_job_template(self):
+        from prefect.workers.utilities import (
+            get_default_base_job_template_for_infrastructure_type,
+        )
+
+        base_job_template = await get_default_base_job_template_for_infrastructure_type(
+            self.get_corresponding_worker_type(),
+        )
+        if base_job_template is None:
+            return super().generate_work_pool_base_job_template()
```

Comment:
Can you talk me through what the situation for hitting this case would be? Would this mean that the `ProcessWorker` doesn't exist locally, so we backoff to a generic work pool template?

Reply:
This doesn't apply to this block since we can derive it from the `ProcessWorker` class in the core library. For the other types, in the unlikely chance that we can't find a default base job template, we can still create a `block` work pool that wraps the block being published.

I'll remove this statement.